### PR TITLE
Add info to list-clouds output to list maas and other such providers

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

juju clouds output is improved to show private cloud providers like maas

## QA steps

$ juju clouds
Cloud        Regions  Default          Type        Description
aws               14  us-east-1        ec2         Amazon Web Services
aws-china          1  cn-north-1       ec2         Amazon China
aws-gov            1  us-gov-west-1    ec2         Amazon (USA Government)
<snip>

Try 'list-regions <cloud>' to see available regions.
'show-cloud <cloud>' or 'regions --format yaml <cloud>' can be used to see region endpoints.
Update the known public clouds with 'update-clouds'.
'add-cloud' can add private clouds or private infrastructure built using:
  - maas, manual, openstack, oracle, vsphere

^^^^^ new

### CODE REVIEW

File: cmd/juju/cloud/list.go [hunk 1]
```diff
@@ -203,10 +204,18 @@ func formatCloudsTabular(writer io.Writer, value interface{}) error {
...
+	providers, _, err := cloudProviders()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
 	w.Println("\nTry 'list-regions <cloud>' to see available regions.")
 	w.Println("'show-cloud <cloud>' or 'regions --format yaml <cloud>' can be used to see region endpoints.")
-	w.Println("'add-cloud' can add private clouds or private infrastructure.")
 	w.Println("Update the known public clouds with 'update-clouds'.")
+	w.Println("'add-cloud' can add private clouds or private infrastructure built using:")
```
Comment (axw) [inline cmd/juju/cloud/list.go]: s/using/for the following provider types/
?

File: cmd/juju/cloud/add.go [hunk 2]
```diff
@@ -264,15 +264,15 @@ func queryName(
 	}
 }
 
-func queryCloudType(pollster *interact.Pollster) (string, error) {
+// cloudProviders returns the names of providers supported by add-cloud,
+// and also the names of those which are not supported.
+func cloudProviders() (providers []string, unsupported []string, _ error) {
```
Comment (axw) [inline cmd/juju/cloud/add.go]: The function name could explain what it returns a bit better. Maybe "addableCloudProviders" (sounds a bit weird, but explains better...)?

File: cmd/juju/cloud/list.go (hunk 1 above)
Comment (wallyworld) [inline cmd/juju/cloud/list.go]: done

File: cmd/juju/cloud/add.go (hunk 2 above)
Comment (wallyworld) [inline cmd/juju/cloud/add.go]: done

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**